src/OAuth2/Storage/IbmDb2.php

```diff
@@ -0,0 +1,606 @@
+<?php
+
+namespace OAuth2\Storage;
+
+use OAuth2\OpenID\Storage\UserClaimsInterface;
+use OAuth2\OpenID\Storage\AuthorizationCodeInterface as OpenIDAuthorizationCodeInterface;
+
+/**
+ *
+ * NOTE: This class is meant to get users started
+ * quickly. If your application requires further
+ * customization, extend this class or create your own.
+ *
+ * NOTE: Passwords are stored in plaintext, which is never
+ * a good idea.  Be sure to override this for your application
+ *
+ * @author Brent Shaffer <bshafs at gmail dot com>
+ * @author Alan Seiden <alan at alanseiden dot com>
+ */
+class IbmDb2 implements
+    AuthorizationCodeInterface,
+    AccessTokenInterface,
+    ClientCredentialsInterface,
+    UserCredentialsInterface,
+    RefreshTokenInterface,
+    JwtBearerInterface,
+    ScopeInterface,
+    PublicKeyInterface,
+    UserClaimsInterface,
+    OpenIDAuthorizationCodeInterface
+{
+    protected $db;
+    protected $config;
+
+    public function __construct($connection, $config = array())
+    {
+        if (!is_resource($connection)) {
+            // Note: Unlike PDO, IbmDb2 (ibm_db2 extension) cannot be configured via dsn string.
+
+            if (!is_array($connection)) {
+                throw new \InvalidArgumentException('First argument to OAuth2\Storage\IbmDb2 must be a resource or a configuration array');
+            }
+
+            // merge optional parameters. Set empty defaults if not present in $connection array.
+            $connection = array_merge(array(
+                'database' =>     '',
+                'username' =>   '',
+                'password' =>   '',
+                'persistent' => false,
+                'driver_options' =>    array(),
+            ), $connection);
+
+            // use persistent or not
+            $isPersistent = $connection['persistent'];
+            $connectFunction = ((bool) $isPersistent) ? 'db2_pconnect' : 'db2_connect';
+
+            // try to connect
+            $connection = $connectFunction($connection['database'], $connection['username'], $connection['password'], $connection['driver_options']);
+
+            // this is how ZF2 handles connection errors
+            if ($connection === false) {
+                throw new Exception\RuntimeException(sprintf(
```

Comment:
This will throw a namespace error, as it will be looking for `OAuth2\Storage\ Exception\RuntimeException`